common/totp.js

```diff
@@ -1,89 +1,26 @@
-import {jsSHA} from "./sha.js";
+import {hmac_sha1} from "./hmac-sha1.js";
 
-export function TOTP() {
-  
-  var dec2hex = function(s) {
-    return (s < 15.5 ? "0" : "") + Math.round(s).toString(16);
-  };
-
-  var hex2dec = function(s) {
-    return parseInt(s, 16);
-  };
-
-  var leftpad = function(s, l, p) {
-    if (l + 1 >= s.length) {
-        s = Array(l + 1 - s.length).join(p) + s;
-    }
-    return s;
-  };
-  
-  var parsehex = function(bits) {
-    let hex = "";
-    for (let i=0; i+4<=bits.length; i+=4) {
-      let chunk = bits.substr(i, 4);
-      hex = hex + parseInt(chunk, 2).toString(16) ;
-    }
-    return hex;
-  };
-
-  var hexpad = function(bits) {
-    // Padding to avoid 'String of HEX type must be in byte increments' error.
-    let i = bits.length;
-    
-    for (i = i % 8; i > 0; i--) {
-      bits += leftpad('0', 5, '0');
-    }
-    return parsehex(bits);
-  };
-  
-  var base32tohex = function(base32) {
-    let base32chars = "ABCDEFGHIJKLMNOPQRSTUVWXYZ234567";
-    let bits = "";
-    let i;
-    
-    base32 = base32.replace(/ +/g, ""); // Strip whitespace
-    
-    for (i=0; i<base32.length; i++) {
-      let char = base32.charAt(i).toUpperCase();
-      if (char === '0') {
-          char = 'O';
-      } else if (char === '8') {
-          char = 'B';
-      } else if (char === '1') {
-          char = 'L';
-      }
-      let val = base32chars.indexOf(char);
-      if ( val >= 0 && val < 32 ) {
-        bits += leftpad(val.toString(2), 5, '0');
-      } else {
-        throw Error("Character out of range: " + char);
-      }
-    }
-    
-    let hex = parsehex(bits);
-    
-    if (hex.length % 2 !== 0) {
```

Comment:
You'll need to add support for the hexpad function/check or an alternative in your new `base32ToUint8Array` function. With your current proposed changes 26 or other odd length tokens no longer work. This issue (#3) was previously solved with the addition of this function.

Reply:
Whoops, sorry. I thought I had covered that, since I had been testing with Amazon tokens, but tokens with 26 characters are definitely broken with my changes. I'll get that sorted and update the branch.

Reply:
Turns out just removing the check for leftover bits allows base32 strings with weird lengths to work. Tested with the 26 char test string from `app/index.js` and some other random-length base32 strings and it seems to decode properly now.